Flick/Controllers/AddToListViewController.swift

```diff
@@ -296,11 +297,21 @@ class AddToListViewController: UIViewController {
         }
     }
 
-    private func searchMediaByQuery(_ query: String) {
-        NetworkManager.searchMedia(query: query) { [weak self] media in
-            guard let self = self, self.isSearching else { return }
-            self.searchResultMedia = media
-            self.resultMediaTableView.reloadData()
+    @objc private func getMedia(timer: Timer) {
+        if let userInfo = timer.userInfo as? [String: String],
+            let searchText = userInfo["searchText"] {
+            NetworkManager.searchMedia(query: searchText) { [weak self] query, media in
+                guard let self = self, self.isSearching else { return }
+                // Update search result only if there's no query or query matches current searchText
+                if let query = query, query != self.searchBar.searchTextField.text {
```

Comment:
this is just an extra check to make sure the current text in searchBar matches the searchText we queried cuz it could be different since we could have made a few queries